lang/cs/Org.Apache.REEF.Wake/Remote/IWritable.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Org.Apache.REEF.Wake.Remote
+{
+    /// <summary>
+    /// Interface that classes should implement if they need to be readable to and writable 
+    /// from the stream. It is assumed that the classes inheriting this interface will have a 
+    /// default empty constructor
+    /// </summary>
+    public interface IWritable
+    {
+        /// <summary>
+        /// Read the class fields from the stream.
+        /// </summary>
+        /// <param name="stream">The stream from which to read</param>
+        /// <param name="optionalParameters">The optional parameters to be passed to the reader.
```

Comment:
In NSMessage we have IIdentifier variables. They need to be written as string in WriteFunction which is okay. But during Read Function string needs to be converted to IIdentifier for which an IIdentifierFactory variable needs to be passed during read time.